pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -148,66 +152,162 @@ function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: st
  * Windows only. Read settings values from registry using schemaObj as a template.
  * @param schemaObj the object used as a template for navigating registry.
  * @param regKey the registry key obtained from nativeReg.openKey().
+ * @param pathParts the path to the current registry key
  * @returns null, or the registry data as an object.
  */
-function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): RecursivePartial<settings.Settings>|null {
+function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY, pathParts: string[]): RecursivePartial<settings.Settings>|null {
   let newObject: RecursivePartial<settings.Settings>|null = null;
 
   const schemaKeys = Object.keys(schemaObj);
+  const schemaKeysLowerCase = schemaKeys.map(k => k.toLowerCase());
+  const commonKeys: string[] = [];
+  const unknownKeys: string[] = [];
+  const specialObjectKeys: string[] = [];
+  let regValue: any;
+
   // ignore case
-  const registryKeys = nativeReg.enumKeyNames(regKey).concat(nativeReg.enumValueNames(regKey)).map(k => k.toLowerCase());
-  const commonKeys = schemaKeys.filter(k => registryKeys.includes(k.toLowerCase()));
+  nativeReg.enumKeyNames(regKey).forEach((k) => {
+    if (k.toLowerCase() === 'integrations' && pathParts.length === 3 && pathParts[2] === 'WSL') {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'extensions' && pathParts.length === 2) {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'mutedchecks' && pathParts.length === 3 && pathParts[2] === 'diagnostics') {
+      specialObjectKeys.push(k);
+    } else if (schemaKeysLowerCase.includes(k.toLowerCase())) {
+      commonKeys.push(k);
+    } else {
+      unknownKeys.push(k);
+    }
+  });
+  if (unknownKeys.length) {
+    console.log(`Unrecognized keys in registry at \\${ pathParts.join('\\') }: [${ unknownKeys.join(', ') }]`);
+  }
 
+  // First process the nested keys, then process any values
   for (const k of commonKeys) {
     const schemaVal = schemaObj[k];
-    let regValue: any = null;
+    const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
```

Comment:
See above — we may attempt to open a value as a key (and fail), and go to the next thing. I _guess_ that's okay? Just seems odd.

Reply:
But I think _this_ comment is mistaken; `k` is the name of a key, so we can open it fine.